### ISSUE TITLE
Couldn't read data for number <multi-account>: EOF. Is the number properly registered?

### ISSUE THREAD

**Issue by ronaldt80** [bbernhard/signal-cli-rest-api#632]
### The problem

I continuously get an error "time="2024-12-23T19:15:22Z" level=error msg="Couldn't read data for number <multi-account>: EOF. Is the number properly registered?"" Subsequently when I try to invoke the api, either through CLI or via Actions in Home Assistant I get broken pipe errors. A restart of the addon solves the issue temporarily but after a while (more or less random) the issue returns.

Read through the earlier filings of this error a tried the following steps:
- Rolled back 1 version
- Added a user name
- Performed the rate limit challenge

So far no luck - I have used the addon for years - this is the first time I faced these issues, any tips?

### Are you using the latest released version?

- [X] Yes

### Have you read the troubleshooting page?

- [X] Yes

### What type of installation are you running?

Home Assistant Signal Addon

### In which mode are you using the docker container?

JSON-RPC Mode

### What's the architecture of your host system?

x86-64

**Comment by ronaldt80** [comment]
I see now my json rpc file includes the following:

```
config:
  <multi-account>:
    tcp_port: 6001
    fifo_pathname: /tmp/sigsocket1
```

is that normal?

**Comment by ronaldt80** [comment]
I manually changed the yml file by replacing "multi-account" by my phone number but it reverts automatically. Non-RPC mode works but the RPC mode throws the error and changes back the yml file

**Comment by ronaldt80** [comment]
The error popped up since I upgraded the home assistant add-on to v 0.9; don't know what exactly triggered it but can send 1 or 2 messages and then it stops- broken pipe etc. I thought it had to do with sending too many messages over a short time span and was signal server related but not sure how to check and seeing it only happens on RPC it's probably not that.

The EOF error subsequently appears every 5 minutes (to the second) in the log. Weird thing is I have been running this without issues for many yrs- last 3 years on RPC also without issues.

Rolling back to 0.89 did not change anything, switching away from RPC did (but it's so slow now 😃).

I by the way use a dedicated number - will probably have to find my SIM card and indeed register. Thanks for suggesting and explaining how (after being without errors for this long I guess I became complacent 😉 - had trouble finding my way to the swagger docs but it quickly came back).

**Comment by bbernhard** [comment]
Did you maybe register another number/link a device in the past that you do not use anymore? (It might be worth checking the contents of the `signal-cli-config/data` folder - which contains configs for all the registered numbers/linked devices).

One difference between the json-rpc and the normal/native mode is, that in json-rpc mode it tries to connect to all registered numbers/linked devices and checks for incoming messages. So, if you have at some point registered another number, it would try to receive messages for that number, although it might not be in use anymore.

**Comment by ronaldt80** [comment]
Thanks, will try that and revert. Would be surprised but who knows..

**Comment by ronaldt80** [comment]
Quite weird.. I tried to re-register yesterday but got stuck on the captcha part. So no luck - but this morning I switched back to JSON RPC - just to see if I could fix what ever issues the addon has, but issues are gone now..

I will leave this open for a few more days and revert later in the week.

**Comment by ronaldt80** [comment]
The issue returned, however I also noticed a couple of other things on my machine. The mariadb addon crashed several times, no real noticeable issue. Someone recommended to increase the memory; so did 2 things on my virtual machine, doubled the number of CPU cores and increased RAM allocation by 50%. Ever since my database issues were resolved but, coincidently (or not) also Signal performs.

Strongly get the feeling this is related to the latest upgrade of HA OS - seeing my machine has been stable for years and not much changed. Who knows..
